src/parser/parser_v2.cpp

```diff
@@ -427,33 +431,39 @@ rule_data_container parse_rule_data(
     for (ddwaf::parameter object : rule_data) {
         std::string id;
         try {
-            ddwaf::parameter::map entry = object;
+            auto entry = static_cast<ddwaf::parameter::map>(object);
 
             id = at<std::string>(entry, "id");
 
+            auto type = at<std::string_view>(entry, "type");
+            auto data = at<parameter>(entry, "data");
+
+            std::string_view operation;
             auto it = rule_data_ids.find(id);
             if (it == rule_data_ids.end()) {
-                DDWAF_DEBUG("Rule data id '%s' has no associated rule", id.c_str());
-                continue;
+                // Infer processor from data type
+                if (type == "ip_with_expiration") {
+                    operation = "ip_match";
+                } else if (type == "data_with_expiration") {
+                    operation = "exact_match";
+                } else {
+                    DDWAF_DEBUG("Failed to process rule idata id '%s", id.c_str());
+                }
+            } else {
+                operation = it->second;
             }
 
-            auto type = at<std::string_view>(entry, "type");
-
-            DDWAF_DEBUG("Updating rules with id '%s' and type '%s'", id.c_str(), type.data());
-
-            auto data = at<parameter>(entry, "data");
-
             rule_processor::base::ptr processor;
-            if (it->second == "ip_match") {
+            if (operation == "ip_match") {
                 using rule_data_type = rule_processor::ip_match::rule_data_type;
                 auto parsed_data = parser::parse_rule_data<rule_data_type>(type, data);
                 processor = std::make_shared<rule_processor::ip_match>(parsed_data);
-            } else if (it->second == "exact_match") {
+            } else if (operation == "exact_match") {
                 using rule_data_type = rule_processor::exact_match::rule_data_type;
                 auto parsed_data = parser::parse_rule_data<rule_data_type>(type, data);
                 processor = std::make_shared<rule_processor::exact_match>(parsed_data);
             } else {
-                DDWAF_WARN("Processor %s doesn't support dynamic rule data", it->second.c_str());
+                DDWAF_WARN("Processor %s doesn't support dynamic rule data", operation.data());
```

Comment:
In general, string_views need not be nul-terminated (depends on the source of the buffer).